pkg/gardenlet/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -327,9 +327,17 @@ func (r *reconciler) delete(ctx context.Context, gardenClient client.Client, con
 		conditionInstalled = gardencorev1beta1helper.UpdatedCondition(conditionInstalled, gardencorev1beta1.ConditionFalse, "DeletionFailed", fmt.Sprintf("Deletion of ManagedResource secret %q failed: %+v", controllerInstallation.Name, err))
 	}
 
-	if err := seedClient.Client().Delete(ctx, getNamespaceForControllerInstallation(controllerInstallation)); client.IgnoreNotFound(err) != nil {
+	namespace := getNamespaceForControllerInstallation(controllerInstallation)
+	err = seedClient.Client().Delete(ctx, namespace)
+	if err == nil || apierrors.IsConflict(err) {
+		message := fmt.Sprintf("Deletion of Namespace %q is still pending.", namespace.Name)
+		conditionInstalled = gardencorev1beta1helper.UpdatedCondition(conditionInstalled, gardencorev1beta1.ConditionFalse, "DeletionPending", message)
+		return errors.New(message)
```

Comment:
If I understand correctly, what we would like to achieve here is to requeue on successful deletion, and keep requeueing until the resource is eventually fully gone ("not found" error). Could we achieve this without reporting the successful deletion as an error, e.g. by adding a `requeue` boolean result to the `delete` method and requeue if it's true? 

I did something similar in the `ManagedSeed` reconciler, see https://github.com/gardener/gardener/blob/7145aa25f9523945ec7bfbd257ec9101176ade4c/pkg/gardenlet/controller/managedseed/managedseed_reconciler.go#L88-L97

Reply:
I adopted the suggestion in 2cc5bbe32b0081018c36f192e30383be98e6eb69.

Reply:
Thanks!